crates/biome_js_analyze/src/assist/source/organize_imports.rs

```diff
@@ -1,144 +1,521 @@
+    /// ```
+    ///
+    /// Note that **source** is also often called **specifier** in the JavaScript ecosystem.
+    ///
+    ///
+    /// ## Chunk of imports and chunks of exports
+    ///
+    /// A **chunk** is a sequence of adjacent imports or exports.
+    /// A chunk contains only imports or exports, not both at the same time.
+    /// The following example includes two chunks.
+    /// The first chunk consists of the three imports and the second chunk consists of the three exports.
+    /// The first `export` that appears ends the first chunk and starts a new one.
+    ///
+    /// ```js,ignore
+    /// import A from "a";
+    /// import * as B from "b";
+    /// import { C } from "c";
+    /// export * from "d";
+    /// export * as F from "e";
+    /// export { F } from "f";
+    /// ```
+    ///
+    /// Chunks also ends as soon as a statement or a **side-effect import** (also called _bare import_) is encountered.
+    /// Each side-effect import consists in a chunk:
+    /// The following example contains six chunks:
+    ///
+    /// 1. The first chunk contains the two first `import` and ends with the appearance of the first side-effect import `import "x"`.
+    /// 2. The second chunk contains only the side-effect import `import "x"`.
+    /// 3. The third chunk contains only the side-effect import `import "y"`.
+    /// 4. The fourth chunk contains a single `import`; The first `export` ends it.
+    /// 5. The fifth chunk contains the first `export`; The function declaration ends it.
+    /// 6. The sixth chunk contains the last two `export`.
+    ///
+    /// ```js,ignore
+    /// // chunk 1
+    /// import A from "a";
+    /// import * as B from "b";
+    /// // chunk 2
+    /// import "x";
+    /// // chunk 3
+    /// import "y";
+    /// // chunk 4
+    /// import { C } from "c";
+    /// // chunk 5
+    /// export * from "d";
+    /// function f() {}
+    /// // chunk 6
+    /// export * as E from "e";
+    /// export { F } from "f";
+    /// ```
+    ///
+    /// Chunks are also delimited by detached comments.
+    /// A **detached comment** is a comment followed by a blank line.
+    /// Comments not followed by a blank line are **attached comments**.
+    /// Note that blank lines alone are not taken into account when chunking imports and exports.
+    /// The following example contains a detached comment that splits the imports into two chunks.
+    /// `import C from "c"` forms the second chunk.
+    /// The other blank lines are ignored: the two first `import` forms the first chunk and the three last `export` form the third chunk.
+    ///
+    /// ```js,ignore
+    /// // Attached comment 1
+    /// import A from "a";
+    ///
+    /// // Attached comment 2
+    /// import * as B from "b";
+    /// // Detached comment
+    ///
+    /// import { C } from "c";
+    /// export * from "d";
+    ///
+    /// export * as F from "e";
+    /// export { F } from "f";
+    /// ```
```

Comment:
```suggestion
    /// The following example contains a detached comment that splits the imports into two chunks:
    ///
    /// ```js,ignore
    /// // Attached comment 1
    /// import A from "a";
    ///
    /// // Attached comment 2
    /// import * as B from "b";
    /// // Detached comment
    ///
    /// import { C } from "c";
    /// ```
    ///
    /// The line `import { C } from "c"` forms the second chunk.
    /// The blank line between the first two imports is ignored so they form a single chunk.
```
(Also removed the `export`s so the example is more focused.)